File: martianrun/core/src/com/gamestudio24/martianrun/actors/menu/GameLabel.java

package com.gamestudio24.martianrun.actors.menu;

import com.badlogic.gdx.graphics.g2d.Batch;
import com.badlogic.gdx.graphics.g2d.BitmapFont;
import com.badlogic.gdx.math.Rectangle;
import com.badlogic.gdx.scenes.scene2d.Actor;
import com.gamestudio24.martianrun.utils.AssetsManager;
import com.gamestudio24.martianrun.utils.Constants;

public class GameLabel extends Actor {

    private final Rectangle bounds;
    private final BitmapFont font;

    public GameLabel(Rectangle bounds) {
        this.bounds = bounds;
        setWidth(bounds.width);
        setHeight(bounds.height);
        font = AssetsManager.getLargeFont();
    }

    @Override
    public void draw(Batch batch, float parentAlpha) {
        super.draw(batch, parentAlpha);
        font.drawWrapped(batch, Constants.GAME_NAME, bounds.x, bounds.y, bounds.width, BitmapFont.HAlignment.CENTER);
    }

}

File: martianrun/core/src/com/gamestudio24/martianrun/actors/menu/GameLabel.kt
package com.gamestudio24.martianrun.actors.menu

import com.badlogic.gdx.graphics.g2d.Batch
import com.badlogic.gdx.graphics.g2d.BitmapFont
import com.badlogic.gdx.math.Rectangle
import com.badlogic.gdx.scenes.scene2d.Actor
import com.gamestudio24.martianrun.utils.AssetsManager.getLargeFont
import com.gamestudio24.martianrun.utils.Constants

class GameLabel(private val bounds: Rectangle) : Actor() {
    private val font: BitmapFont

    init {
        setWidth(bounds.width)
        setHeight(bounds.height)
        font = getLargeFont()
    }

    override fun draw(batch: Batch?, parentAlpha: Float) {
        super.draw(batch, parentAlpha)
        font.drawWrapped(batch, Constants.GAME_NAME, bounds.x, bounds.y, bounds.width, BitmapFont.HAlignment.CENTER)
    }
}
